modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc

```diff
@@ -0,0 +1,458 @@
+<?php
+/**
+ * This page implements a front end which is used to unify all
+ * of the different MRI violation types (could not identify scan,
+ * header violation in the scan after identifying scan type, mislabeled 
+ * scan, etc) and link to the appropriate module to further investigate
+ * why a scan was excluded by the imaging pipeline scripts.
+ *
+ * PHP Version 5
+ *
+ * @category MRI
+ * @package  Main
+ * @author   Dave MacFarlane <driusan@bic.mni.mcgill.ca>
+ * @license  GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+require_once 'NDB_Menu_Filter.class.inc';
+require_once 'NDB_Menu_Filter_Form.class.inc';
+/**
+ * Implement the menu filter for searching through all violated scans
+ *
+ * @category MRI
+ * @package  Main
+ * @author   Dave MacFarlane <driusan@bic.mni.mcgill.ca>
+ * @license  GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */      
+class NDB_Menu_Filter_Form_mri_violations extends NDB_Menu_Filter_Form
+{
+    /**
+     * Check if user should be allowed to see this page.
+     *
+     * @return boolean true if the user is permitted to see violated scans
+     */
+    function _hasAccess()
+    {
+        // create user object
+        $user =& User::singleton();
+        if (Utility::isErrorX($user)) {
+            return PEAR::raiseError("User Error: ".$user->getMessage());
+        }
+        $this->tpl_data['violated_scans_modifications'] 
+            = $user->hasPermission('violated_scans_edit');
+        return ($user->hasPermission('violated_scans_view_allsites'));
+    }
+
+    function _process($values)
+    {
+        if (!is_array($values) || count($values) ==0) {
+            return true;
+        }
+        $DB =& Database::singleton();
+        $user =& User::singleton();
+        if (Utility::isErrorX($user)) {
+            return PEAR::raiseError("User Error: ".$user->getMessage());
+        }
+        foreach ($values['resolvable'] AS $key=>$val) {
+            $hash = $key;
+            $row = $DB->pselectRow(
+                    "SELECT * FROM violations_resolved
+                    WHERE hash = :hash", array('hash' => $hash));
+            
+            //if row is found it means that resolve status was previously assigned,
```

Comment:
Can these be combined into a single $DB->replace() call to simplify this if the hash was made the primary key?